### PR TITLE
RU: Effective TensorFlow 2.0

### PR DESCRIPTION
**Pull request by 0101011**

### CODE REVIEW

**Comment by lamberta** [comment]
Hi @emedvedev
When you get a chance, would you mind reviewing this translation from Andrew?

Let me know if you run into permission problems, we're still trying to work within GitHub's model. We don't yet have a style guide for Russian terms, but Andrew and @ezhulenev have worked out a few terms in some [earlier pull requests](https://github.com/tensorflow/docs/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed+label%3Atranslation++%22ru%22+).

I started a [translations wiki page](https://github.com/tensorflow/docs/wiki/Translation-notes), but that's a pain. Maybe we can move this to a `README.md` at the root of `/site/ru`?

Thanks for the translation, Andrew, and thanks for volunteering to review, Ed.

**Comment by emedvedev** [comment]
Hi there! I'm going to review in detail shortly, but for now there's one major issue I have with the tone of the translation, which has probably been discussed before, but I still wanted to bring it up.

"You" is translated as "ты" instead of "вы" in this PR (and all the other PRs, as far as I can tell). While this is technically correct, it is highly informal ("ты" is a singular form of _you_ that is normally reserved for familiar people) and generally not suitable for technical documentation. 

If this is done on purpose—for instance, to create an informal and, for the lack of a better term, somewhat playful tone—let me know, and I'll accept this. Otherwise I'd either change to "вы" (a more formal "you") or, ideally, rewrite passive voice, which is much more acceptable for technical docs in Russian, but may steer a bit far from the original.

**Comment by lamberta** [comment]
Thanks for looking and this is a good observation.
Google has a [Developer Documentation Style Guide](https://developers.google.com/style/) that I would prefer we adhere to. There is a section on [style and tone](https://developers.google.com/style/tone):
> Aim, in your documents, for a voice and tone that's conversational, friendly, and respectful without being overly colloquial or frivolous; a voice that's casual and natural and approachable, not pedantic or pushy. Try to sound like a knowledgeable friend who understands what the developer wants to do.

And I understand this doesn't always translate across cultural contexts so I'll defer to you guys. There may be other Google developer documentation sites that could guide us, but I've never read the Russian versions :)

**Comment by ezhulenev** [comment]
I had the same concern, to me `вы` sounds more familiar, and reading `ты` feels really weird. I would agree that `вы` + passive voice is the most expected style from tech docs. Though this informal style is also a totally valid option. Maybe we can do a poll, translate two ~small paragraphs in different styles and ask people what style they'd prefer?

**Comment by 0101011** [comment]
Hi Ed,

1. Thanks for bringing it up! Interestingly enough, we didn't pay attention to it it the previous docs. I will explain the rationale behind usage of the informal style in the tutorials after your review.

2. We've discussed all the sticking points in the previous PRs with Eugene in Russian, but I figure it'll be more productive for us to discuss and propose changes in English so that Mark and Billy could direct us as needed.

3. Feel free to throw new ideas, submit changes and also pay attention to the vocabulary we use: in the previous guides and notebooks Eugene proposed to correct significant vocabulary errors in math and ML which otherwise might have been missed (thank you, @ezhulenev!)

File: site/ru/guide/effective_tf2.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+Раньше побочным продуктом eager execution был метод `tf.control_dependencies()`,
+который теперь не треубется, и все строки кода будут исполняться в последовательном
+порядке, определенном `tf.function`. Сторонний код будет выполняться в 
+определенном ему порядке.
+
+### Никаких глобальных переменных
+
+TensorFlow 1.X внутренне полагался на глобальные переменные. Когда ты вызывал
+`tf.Variable()`, то эта переменная помещалась в стандартный граф и оставалась там
+даже если уже было изменено имя, присвоенное этой переменной. Ты мог восстановить
```
Comment (0101011) [inline site/ru/guide/effective_tf2.md]: `Мы могли восстановить эту переменную` as a better pronoun usage (commenting it here lest I forget)

Comment (0101011) [reply site/ru/guide/effective_tf2.md]: `Вы могли восстановить эту переменную`

File: site/ru/guide/effective_tf2.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
+# Как использовать TensorFlow 2.0?
+
+В TensorFlow 2.0 были сделаны несколько изменений, которые позволят более продуктивно работать всем
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > Были сделаны

Было сделано

> позволят более продуктивно работать всем пользователям TensorFlow

Maybe "позволят всем пользователям TensorFlow работать более продуктивно"?

File: site/ru/guide/effective_tf2.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
+# Как использовать TensorFlow 2.0?
+
+В TensorFlow 2.0 были сделаны несколько изменений, которые позволят более продуктивно работать всем 
+пользователям TensorFlow. В TensorFlow 2.0 были удалены
+[ненужные APIs](https://github.com/tensorflow/community/blob/master/rfcs/20180827-api-names.md),
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > APIs

API

File: site/ru/guide/effective_tf2.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+существующие и используемые API были проработаны 
+([объединены RNN](https://github.com/tensorflow/community/blob/master/rfcs/20180920-unify-rnn-interface.md),
+[а также оптимизаторы](https://github.com/tensorflow/community/blob/master/rfcs/20181016-optimizer-unification.md)),
+улучшена интеграция с рабочей средой Python в режиме
+[Eager execution](https://www.tensorflow.org/guide/eager).
+
+Во многих
+[запросах RFC](https://github.com/tensorflow/community/pulls?utf8=%E2%9C%93&q=is%3Apr)
+объяснялись основные изменения, которые затронут TensorFlow 2.0. В этом документе
+будет показан, как должен выглядеть процесс работы с новым TensorFlow 2.0.
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > будет показан

Будет показано

File: site/ru/guide/effective_tf2.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+Подразумевается, что ты уже знаком с TensorFlow 1.x.
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > ты уже знаком

See above: "вы уже знакомы" instead of "ты".

File: site/ru/guide/effective_tf2.md [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+
+TensorFlow 1.X внутренне полагался на глобальные переменные. Когда ты вызывал
+`tf.Variable()`, то эта переменная помещалась в стандартный граф и оставалась там
+даже если уже было изменено имя, присвоенное этой переменной. Ты мог восстановить
+эту переменную `tf.Variable`, но только если тебе было известно имя, которое было
+присвоено ей при создании. В результате все механизмы были направлены на то, чтобы
+помочь пользователю отыскать имя их переменных еще раз: `tf.get_global_step()`, 
+`tf.global_variables_initializer()`, оптимизаторы, которые рассчитывали
+градиенты по всем обучаемым перменным и так далее. В TensorFlow 2.0 были устранены
+все эти механизмы ([Переменные 2.0 RFC](https://github.com/tensorflow/community/pull/11))
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > В TensorFlow 2.0 были устранены все эти механизмы

I'd suggest reordering: "В TensorFlow 2.0 эти механизмы были устранены"

File: site/ru/guide/effective_tf2.md [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+
+Требование следить за переменными создает дополнительную нагрузку на пользователя,
+но в случае с объектами Keras (см. ниже), это нагрузка - минимальная.
+
+### Функции, а не сессии
+
+Вызов `session.run()` работал почти как вызов функции: ты определял вводные данные
+и вызывал функцию, получая на выходе результаты. В TensorFlow 2.0, ты можешь
+декорировать функцию Python при помощи `tf.function()`, отметив ее как JIT
+компиляцию, чтобы TensorFlow запустил ее на одном единственном графе
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: "Отметив её как JIT-компилируемую"

File: site/ru/guide/effective_tf2.md [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+([Функции 2.0 RFC](https://github.com/tensorflow/community/pull/20)). Этот
+механизм позволяет TensorFlow 2.0 получить все преимущества режима graph:
+
+-   Производительность: функции могут быть оптимизированы (отсечение узлов графа,
+    слияние ядра и так далее)
+-   Портативность: The function can be exported/reimported функции могут быть экспортированы
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > The function can be exported/reimported

Cleanup needed

File: site/ru/guide/effective_tf2.md [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+    или импортированы повторно ([Сохранение моделей 2.0 RFC](https://github.com/tensorflow/community/pull/34)),
+    позволяя пользователям использовать и делиться модульными функциями TensorFlow.
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > использовать и делиться модульными функциями TensorFlow

"Использовать модульные функции TensorFlow и делиться ими"?

File: site/ru/guide/effective_tf2.md [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+
+## Рекомендации и идиомы TensorFlow 2.0
+
+Смотри все примеры использования
+[MNIST (стандартный пример)](../tutorials/beginner/tf2_overview.ipynb)
+
+### Рефакторинг кода на малые функции
+
+Часто используемый шаблон использования в TensorFlow 1.X работал по принципу
+"кухонной раковины", когда быа выложена совокупность всех возможных вычислений,
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: Ideally leave the English phrase as is, `принцип "кухонной раковины"` is not used in Russian and would cause confusion.

File: site/ru/guide/effective_tf2.md [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+а затем выбранные тензоры вычислялись с помощью `session.run()`. В TensorFlow 2.0
+пользователи должны сами разбивать код на более мелкие функции и вызывать каждую
+когда это необходимо. Необязательно декорировать каждую из этих небольших функций
+с `tf.function`; используй `tf.function` для дкеорации только высокоуровневых
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > дкеорации

Tiny typo, "декорации" (or "декорирования")

File: site/ru/guide/effective_tf2.md [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+вычислений - например, один шаг обучения или прямого прохода для модели.
+
+### Используй слои и модели Keras для управления переменными
+
+Keras models and layers offer the convenient `.variables` property, which
+recursively gather up all dependent variables. This makes it very easy to manage
+variables locally to where they are being used.
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: Cleanup

File: site/ru/guide/effective_tf2.md [hunk 13]
```diff
@@ -0,0 +1,311 @@
...
+Модели и слои Keras наследуются из `tf.train.Checkpointable` и интегрируются с
+`@tf.function`, что делает возможным сохранить или непосредственно экспортировать
+сохраненные модели из объектов Keras. Необязательно использовать метод `.fit()` из
+Keras API для этих интеграций.
+
+Here's a transfer learning example that demonstrates how Keras makes it easy to
+collect a subset of relevant variables. Let's say you're training a multi-headed
+model with a shared trunk:
+
+Вот пример переноса обучения (transfer learning), который демонстрирует как легко
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: "Transfer learning" is usually left as is, not translated.

File: site/ru/guide/effective_tf2.md [hunk 14]
```diff
@@ -0,0 +1,311 @@
...
+собрать подмножество необходимых переменных с помощью Keras. Предположим, мы обучаем
+модель разветвленную (multi-head) модель с общим корнем (trunk):
```
Comment (emedvedev) [inline site/ru/guide/effective_tf2.md]: > модель разветвленную (multi-head) модель

A typo :) "Разветвленную (multi-head) модель"

**Comment by lamberta** [comment]
Ok, first, thank you Ed and Eugene for reviewing, and, of course, thank you Andrew for all the doc work you're doing.

After the conversation here and my own in-person conversations with some Russian-speaking Googlers, the more formal form seems preferred. Although we use the "Google Voice" tone where appropriate, we defer to the cultural context for localization.

I don't think it's necessary to make that change in this pull request given the style of the other guides—but you are welcome to. But if someone would like to make a follow-up PR, that would be welcome.

One thing that surprised me in my research is that, in general (meaning outside of our work here), English to Russian technical translations seem rather poor. I guess it surprises me because the Russian folks are the most enthusiastic for TensorFlow translations :) Maybe it's the overlap with machine learning interest. Regardless, if we can sustain interest, I think we might be able to make a rather nice example for Google and beyond.

But one thing is clear, we need a better style guide for translations. I had started a [wiki page](https://github.com/tensorflow/docs/wiki/Translation-notes) for this, but it's difficult to contribute to. How about I just create site/ru/README.md and then we can submit pull requests to that and discuss in the comments.

If anyone has good examples of English->Russian technical translations (preferably Google), let me know and maybe we can use that for guidance. There's the Android developer docs: https://developer.android.com/guide/?hl=ru
But someone mentioned it sounded "passive" and used the formal second person. Passive in English is generally negative so I wasn't sure what that meant.

I'll create a new pull request for the ru/README and I can start a thread on docs@tensorflow.org to further discussion.

Thanks, all. At least we know there's an audience!

And a bonus for me: I'm learning a lot more about the Russian language than I ever thought I'd need!

**Comment by lamberta** [comment]
Hi Andrew, when you get a chance, can you update this pull request?

There's now a [site/ru/README.md](https://github.com/tensorflow/docs/blob/master/site/ru/README.md) and a section to add to Russian style guide. This should be a good place to discuss style and come to agreements.

For future updates we can use the more formal tone—either neutral or slightly friendly—but perhaps not a requirement for this pull request since it matches the other docs you've translated.

Thanks,
Billy

**Comment by 0101011** [comment]
Hey Billy,

I’ll do some housekeeping here as well as reflecting new changes this weekend 👍

**Comment by lamberta** [comment]
Sounds good. Thanks, Andrew.
Infra now in place so expecting to publish the other guides to tensorflow.org soon 🎉

**Comment by 0101011** [comment]
Sounds great! 
Means the docs will soon start being indexed by search engines - we've got quite an interest in DL space in Runet communities.

**Comment by lamberta** [comment]
@0101011 Hey Andrew, what's the status of this notebook? Thanks

**Comment by 0101011** [comment]
Hi Billy, 

I’ll update the PR in 12 hours, send you a notif as soon as it’s ready 👍

**Comment by lamberta** [comment]
Thanks, Andrew. No rush, just checking in :)

File: site/ru/guide/effective_tf2.md (hunk 10 above)
Comment (0101011) [reply site/ru/guide/effective_tf2.md]: Agree!

**Comment by lamberta** [comment]
Hi all, I'd like to get this up by Dev Summit next week, if possible. When @emedvedev's comments have been addressed, can you add your review approval and comment tag me?
Thanks!

**Comment by 0101011** [comment]
Hi Billy, expect a notification within couple of hours 👍

**Comment by 0101011** [comment]
@lamberta The doc is ready to pull Doc, though I guess we'll open a couple of PRs to introduce better wording in some sentences (I see one or two weak spots for polishing)
